dspace-server-webapp/src/test/java/org/dspace/app/rest/AuthorizationFeatureRestRepositoryIT.java

```diff
@@ -0,0 +1,169 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import static com.jayway.jsonpath.JsonPath.read;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.fail;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.jayway.jsonpath.matchers.JsonPathMatchers;
+import org.dspace.app.rest.authorization.AlwaysTrueFeature;
+import org.dspace.app.rest.authorization.AuthorizationFeature;
+import org.dspace.app.rest.authorization.AuthorizationFeatureService;
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Test suite for the Authorization Feature endpoint
+ * 
+ * @author Andrea Bollini (andrea.bollini at 4science.it)
+ *
+ */
+public class AuthorizationFeatureRestRepositoryIT extends AbstractControllerIntegrationTest {
+    @Autowired
+    private AuthorizationFeatureService authzFeatureService;
+
+    @Test
+    /**
+     * All the features should be returned
+     *
+     * @throws Exception
+     */
+    public void findAllTest() throws Exception {
+        int featuresNum = authzFeatureService.findAll().size();
+        int expReturn = featuresNum > 20 ? 20 : featuresNum;
+        String adminToken = getAuthToken(admin.getEmail(), password);
```

Comment:
These tests in this IT all look good, but they all assume you already have admin permissions. Shouldn't we have at least a few tests here to verify the proper errors are thrown if you are unauthenticated or logged in as a non-admin?

Reply:
As these use cases where very quick to test (1 line of code) they are tested at the end of the method (also for the findOne and the search methods) to avoid to slow the test phase without adding to much value. If you thing that is better to have them in a dedicated method I will do. The current test are here
https://github.com/DSpace/DSpace/pull/2663/files#diff-96e7d07bd9f65de53e27a7847cecc571R57
https://github.com/DSpace/DSpace/pull/2663/files#diff-96e7d07bd9f65de53e27a7847cecc571R59

https://github.com/DSpace/DSpace/pull/2663/files#diff-96e7d07bd9f65de53e27a7847cecc571R117
https://github.com/DSpace/DSpace/pull/2663/files#diff-96e7d07bd9f65de53e27a7847cecc571R120

https://github.com/DSpace/DSpace/pull/2663/files#diff-96e7d07bd9f65de53e27a7847cecc571R129
https://github.com/DSpace/DSpace/pull/2663/files#diff-96e7d07bd9f65de53e27a7847cecc571R131

https://github.com/DSpace/DSpace/pull/2663/files#diff-96e7d07bd9f65de53e27a7847cecc571R160
https://github.com/DSpace/DSpace/pull/2663/files#diff-96e7d07bd9f65de53e27a7847cecc571R164

Reply:
@abollini : could you please just add a comment at the end of `findAll()` to say something like `"Non-admin tests are performed below in ____  (reference the methods where they are done)"`.  It makes it easier for others to realize that you purposefully chose not to repeat the same tests -- otherwise, I tend to assume you may have accidentally forgotten them.

Reply:
I have added comments to note which scenario is tested from time to time (admin, normal user, anonymous). As it is the line below your comment it is not shown as outdated